packages/interactivity-router/src/head.ts

```diff
@@ -41,60 +57,59 @@ export const updateHead = async ( newHead: HTMLHeadElement[] ) => {
  * Fetches and processes head assets (stylesheets and scripts) from a specified document.
  *
  * @async
- * @param doc               The document from which to fetch head assets. It should support standard DOM querying methods.
- * @param headElements      A map of head elements to modify tracking the URLs of already processed assets to avoid duplicates.
- * @param headElements.tag
- * @param headElements.text
+ * @param doc The document from which to fetch head assets. It should support standard DOM querying methods.
  *
  * @return Returns an array of HTML elements representing the head assets.
  */
 export const fetchHeadAssets = async (
-	doc: Document,
-	headElements: Map< string, { tag: HTMLElement; text: string } >
+	doc: Document
 ): Promise< HTMLElement[] > => {
 	const headTags = [];
-	const assets = [
-		{
-			tagName: 'style',
-			selector: 'link[rel=stylesheet]',
-			attribute: 'href',
-		},
-		{ tagName: 'script', selector: 'script[src]', attribute: 'src' },
-	];
-	for ( const asset of assets ) {
-		const { tagName, selector, attribute } = asset;
-		const tags = doc.querySelectorAll<
-			HTMLScriptElement | HTMLStyleElement
-		>( selector );
 
-		// Use Promise.all to wait for fetch to complete
-		await Promise.all(
-			Array.from( tags ).map( async ( tag ) => {
-				const attributeValue = tag.getAttribute( attribute );
-				if ( ! headElements.has( attributeValue ) ) {
-					try {
-						const response = await fetch( attributeValue );
-						const text = await response.text();
-						headElements.set( attributeValue, {
-							tag,
-							text,
-						} );
-					} catch ( e ) {
-						// eslint-disable-next-line no-console
-						console.error( e );
-					}
-				}
+	const scripts = doc.querySelectorAll< HTMLScriptElement >(
+		'script[type="module"][src]'
+	);
 
-				const headElement = headElements.get( attributeValue );
-				const element = doc.createElement( tagName );
-				element.innerText = headElement.text;
-				for ( const attr of headElement.tag.attributes ) {
-					element.setAttribute( attr.name, attr.value );
+	Array.from( scripts ).forEach( ( script ) => {
+		const src = script.getAttribute( 'src' );
+		if ( ! headElements.has( src ) ) {
+			// add the <link> elements to prefetch the module scripts
+			const link = doc.createElement( 'link' );
+			link.rel = 'modulepreload';
+			link.href = src;
+			document.head.append( link );
+			headElements.set( src, { tag: script } );
+		}
+	} );
+
+	const stylesheets = doc.querySelectorAll< HTMLScriptElement >(
+		'link[rel=stylesheet]'
+	);
+
+	await Promise.all(
+		Array.from( stylesheets ).map( async ( tag ) => {
+			const href = tag.getAttribute( 'href' );
```

Comment:
This is nullable… there should probable be an early return if there's no `href`:

```suggestion
			const href = tag.getAttribute( 'href' );
			if ( ! href ) {
				return;
			}
```

We need to remove `strict: false` from this project's tsconfig.

Reply:
Done in cf73a668fb5cae25511b51eaf45638164cb4f08e